DNN Platform/DotNetNuke.Abstractions/Logging/IEventLogConfigService.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.Abstractions.Logging
+{
+    using System.Collections;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// The event log config service provides APIs for managing the
+    /// Event Log configuration and individual logs itself.
+    /// </summary>
+    public interface IEventLogConfigService
+    {
+        /// <summary>
+        /// Adds the log type.
+        /// </summary>
+        /// <param name="configFile">The configuration file.</param>
+        /// <param name="fallbackConfigFile">The fallback configuration file.</param>
+        void AddLogType(string configFile, string fallbackConfigFile);
+
+        /// <summary>
+        /// Adds the log type config info.
+        /// </summary>
+        /// <param name="logTypeConfig">The log type configuration.</param>
+        void AddLogTypeConfigInfo(ILogTypeConfigInfo logTypeConfig);
+
+        /// <summary>
+        /// Adds the log.
+        /// </summary>
+        /// <param name="logType">The log type.</param>
+        void AddLogType(ILogTypeInfo logType);
+
+        /// <summary>
+        /// Clears the log.
+        /// </summary>
+        void ClearLog();
+
+        /// <summary>
+        /// Deletes the log.
+        /// </summary>
+        /// <param name="logInfo">the log info.</param>
+        void DeleteLog(ILogInfo logInfo);
+
+        /// <summary>
+        /// Deletes the log type.
+        /// </summary>
+        /// <param name="logType">The log type.</param>
+        void DeleteLogType(ILogTypeInfo logType);
+
+        /// <summary>
+        /// Delete the log type config info.
+        /// </summary>
+        /// <param name="logTypeConfig">The log type config.</param>
+        void DeleteLogTypeConfigInfo(ILogTypeConfigInfo logTypeConfig);
+
+        /// <summary>
+        /// Get the logs.
+        /// </summary>
+        /// <param name="portalID">The portal id.</param>
+        /// <param name="logType">The log type.</param>
+        /// <param name="pageSize">The page size.</param>
+        /// <param name="pageIndex">The page index.</param>
+        /// <param name="totalRecords">The total records.</param>
+        /// <returns>The logs.</returns>
+        IEnumerable<ILogInfo> GetLogs(int portalID, string logType, int pageSize, int pageIndex, ref int totalRecords);
```

Comment:
I think this should be on `IEventLogger`.

Or a new interface, e.g. `IEventLogService`, could have a more complete CRUD interface, while `IEventLogger` is focused on the 99% use case of just logging stuff.  Not sure what's the right conceptual limit of interfaces.

Reply:
This might be the direction to go

Reply:
This has been updated in the latest code push, thanks for the feedback